src/extension/infer.rs

```diff
@@ -323,10 +323,11 @@ impl UnificationContext {
         for tgt_node in hugr.nodes() {
             let sig: &OpType = hugr.get_nodetype(tgt_node).into();
             // Incoming ports with a dataflow edge
-            for port in hugr
-                .node_inputs(tgt_node)
-                .filter(|src_port| matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_))))
-            {
+            for port in hugr.node_inputs(tgt_node).filter(|src_port| {
+                matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_)))
+                    || matches!(sig.port_kind(*src_port), Some(EdgeKind::Static(_)))
+                    || matches!(sig.port_kind(*src_port), Some(EdgeKind::ControlFlow))
```

Comment:
Adding `ControlFlow` edges as a generator of constraints is kind of irrelevant to this PR, just a drive-by

Reply:
I'd strongly suggest moving this into #529. It makes that more self-contained, more obviously complete (!), and it's really not relevant here

Reply:
Also, can you do `matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_)) | Some(EdgeKind::Static(_)) | Some(EdgeKind:ControlFlow))`? That's possible in normal patterns (outside `matches!`), so maybe it works there too...

Reply:
Removed the ControlFlow match